src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/DepedencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,103 @@
+
+        /// <inheritdoc/>
+        public string ListenerName { get; } = "HttpHandlerDiagnosticListener"; // This value comes from: https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/System/Net/Http/HttpHandlerLoggingStrings.cs#L12
+
+        public DependencyCollectorDiagnosticListener(TelemetryClient client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Request")]
+        public void OnRequestSent(HttpRequestMessage request, Guid loggingRequestId, long timestamp)
+        {
+            string httpMethod = request.Method.Method;
+            Uri requestUri = request.RequestUri;
+            string resourceName = requestUri.AbsolutePath;
+            if (!string.IsNullOrEmpty(httpMethod))
+            {
+                resourceName = httpMethod + " " + resourceName;
+            }
+
+            DependencyTelemetry telemetry = new DependencyTelemetry();
+            telemetry.Start();
+            this.client.Initialize(telemetry);
+
+            telemetry.Name = resourceName;
+            telemetry.Target = requestUri.Host;
+            telemetry.Type = "Http";
+            telemetry.Data = requestUri.OriginalString;
+
+            // Add the source instrumentation key header if collection is enabled, the request host is not in the excluded list and the same header doesn't already exist
+            if (!string.IsNullOrEmpty(telemetry.Context.InstrumentationKey) && !request.Headers.Contains(RequestResponseHeaders.SourceInstrumentationKeyHeader))
+            {
+                request.Headers.Add(RequestResponseHeaders.SourceInstrumentationKeyHeader, InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(telemetry.Context.InstrumentationKey));
+            }
+
+            // Add the root ID
+            string rootId = telemetry.Context.Operation.Id;
+            if (!string.IsNullOrEmpty(rootId) && !request.Headers.Contains(RequestResponseHeaders.StandardRootIdHeader))
+            {
+                request.Headers.Add(RequestResponseHeaders.StandardRootIdHeader, rootId);
+            }
+
+            // Add the parent ID
+            string parentId = telemetry.Id;
+            if (!string.IsNullOrEmpty(parentId) && !request.Headers.Contains(RequestResponseHeaders.StandardParentIdHeader))
+            {
+                request.Headers.Add(RequestResponseHeaders.StandardParentIdHeader, parentId);
+            }
+
+            this.requestTelemetry.TryAdd(loggingRequestId, telemetry);
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Response' event.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Response")]
+        public void OnResponseReceived(HttpResponseMessage response, Guid loggingRequestId, long timestamp)
```

Comment:
The actual call to send this request goes through [here](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/netcore50/System/Net/HttpHandlerToFilter.cs#L52), and it doesn't look like it throws. This would tell me that OnResponseReceived() would always get called after OnRequestSent().